scripts/build_upload.py

```diff
@@ -0,0 +1,476 @@
+    print()
+    print(bright(red("!!! Pre-checks failed. The BUILD and UPLOAD has been aborted.")))
+    print()
+    print(bright(red("!!! NO ASSETS HAVE BEEN UPLOADED")))
+    print()
+    banner(red, "{:^80}".format("Bokeh %r build and upload: FAILURE" % CONFIG.version))
+    sys.exit(1)
+
+def abort_builds():
+    print(red("\n!!! FATAL problems occurred during BUILDS"))
+    print()
+    print(bright(red("!!! NO ASSETS HAVE BEEN UPLOADED")))
+    print()
+    print(bright(yellow("Here is the status of all build steps:")))
+    print()
+    for build in CONFIG.builds:
+        print("    - %-10s: %s" % (build, CONFIG.build_status[build]))
+    print()
+    banner(red, "{:^80}".format("Bokeh %r build and upload: FAILURE" % CONFIG.version))
+    sys.exit(1)
+
+def abort_uploads():
+    print(red("\n!!! FATAL problems occurred during UPLOADS"))
+    print()
+    print(bright(red("!!! SOME ASSETS MAY HAVE BEEN UPLOADED")))
+    print()
+    print(bright(yellow("Here is the status of all uploads:")))
+    print()
+    for upload in CONFIG.uploads:
+        print("    - %-10s: %s" % (upload, CONFIG.upload_status[upload]))
+    print()
+    banner(red, "{:^80}".format("Bokeh %r build and upload: FAILURE" % CONFIG.version))
+    sys.exit(1)
+
+#--------------------------------------
+#
+# Check functions
+#
+#--------------------------------------
+
+def check_environment_var(name, message):
+    if name in os.environ:
+        passed("Found %s (%s)" % (message, name))
+    else:
+        failed("Missing %s (%s)" % (message, name))
+        abort_checks()
+
+def check_anaconda_creds():
+    try:
+        token = os.environ['ANACONDA_TOKEN']
+        out = run("anaconda -t %s whoami" % token, silent=True)
+        if "Anonymous User" in out:
+            failed("Could NOT verify Anaconda credentials")
+            abort_checks()
+        passed("Verified Anaconda credentials")
+    except Exception as e:
+        failed("Could NOT verify Anaconda credentials")
+        abort_checks()
+
+def check_cdn_creds():
+    try:
+        username = os.environ['RSUSER']
+        key = os.environ['RSAPIKEY']
+        buf = BytesIO()
+        c = pycurl.Curl()
+        c.setopt(c.CAINFO, certifi.where())
+        c.setopt(c.URL, 'https://identity.api.rackspacecloud.com/v2.0/tokens/')
+        c.setopt(c.HTTPHEADER, ['Content-type: application/json'])
+        c.setopt(c.POSTFIELDS, json.dumps({
+            "auth" : {
+                "RAX-KSKEY:apiKeyCredentials" : {
+                    "username" : username,
+                    "apiKey"   : key,
+                }
+            }
+        }))
+        c.setopt(c.WRITEDATA, buf)
+        c.perform()
+        c.close()
+
+        data = json.loads(buf.getvalue().decode())
+
+        cdn_token = data["access"]["token"]["id"]
+        cdn_id = [
+            d["endpoints"][0]["tenantId"]
+            for d in data["access"]["serviceCatalog"]
+            if d["name"] == "cloudFiles"
+        ][0]
```

Comment:
I am saying that I experienced in the past a change in the data structure that broke the script... but it was an API change from them. Pretty sure things will not change in a test run. But could change from one month to another.